sphinx/texinputs/sphinx.sty

```diff
@@ -161,12 +161,78 @@
   \end{minipage}%
   }%
 }
-\def\FrameCommand{\mycolorbox}
+\def\MidFrameCommand{\mycolorbox}
+\let\LastFrameCommand\MidFrameCommand
+
+% We now customize \FrameCommand (for non split frames) and \FirstFrameCommand
+% (split frames), in order for the framed environment in order for it to
+% handle a Title above the frame, which can not be separated by a pagebreak.
+\newcommand*\MyFrameTitle {}
+\newcommand*\MyVerbatimTitle {}
+% cf visit_caption(self, node) in sphinx/writers/latex.py
+\newcommand*\setupcaptionforverbatim [2]
+{% make the \smallskip customizable ?
+    \def\MyVerbatimTitle{\captionof{#1}{#2}\smallskip }%
+}
+
+% \MyCustomFBox is copied from framed.sty's \CustomFBox, but
+% #1=title/caption is to be set _above_ the top rule, not _below_
+% #1 must be "vertical material", it may be left empty.
+
+% The amsmath patches \stepcounter to inhibit stepping under \firstchoice@false
+% We use that too, because framed.sty typesets multiple times its contents,
+\newif\if@MyFirstFramedPass
+
+\long\def\MyCustomFBox#1#2#3#4#5#6#7{%
+  % we set up amsmath (amstext.sty) conditional to inhibit counter stepping
+  % except in first pass
+  \if@MyFirstFramedPass\firstchoice@true
+                  \else\firstchoice@false\fi
+  \leavevmode\begingroup
+  \setbox\@tempboxa\hbox{%
+    \color@begingroup
+      \kern\fboxsep{#7}\kern\fboxsep
+    \color@endgroup}%
+  \hbox{%
+    \lower\dimexpr#4+\fboxsep+\dp\@tempboxa\hbox{%
+      \vbox{%
+        #1% TITLE
+        \hrule\@height#3\relax
+        \hbox{%
+          \vrule\@width#5\relax
+          \vbox{%
+            \vskip\fboxsep
+            \copy\@tempboxa
+            \vskip\fboxsep}%
+          \vrule\@width#6\relax}%
+        #2%
+        \hrule\@height#4\relax}%
+    }%
+  }%
+  \endgroup
+  \global\@MyFirstFramedPassfalse
+}
+
+\def\FrameCommand{% used for non-split frames
+   % this is inspired from framed.sty v 0.96 2011/10/22 lines 185--190
+   % \fcolorbox (see \mycolorbox above) from color.sty uses \fbox.
+   \def\fbox{\MyCustomFBox{\MyFrameTitle}{}%
+                 \fboxrule\fboxrule\fboxrule\fboxrule}%
+   % \fcolorbox from xcolor.sty may use rather \XC@fbox.
+   \let\XC@fbox\fbox
+   \mycolorbox
+}
+% first portion of split frames:
+\let\FirstFrameCommand\FrameCommand
 
+% Unneeded %'s after control words removed.
```

Comment:
Could you remove this line? This is not a description of macros.
Please note the description of codes (cf. purpose, behaviors and how it works).